File: frontend/src/lib/api/issues.ts
/**
 * Issues API - Issue reporting endpoints
 */

import { get, post, put, buildQueryString } from "./helpers";

export interface IssueCategory {
  id: string;
  name: string;
  description: string;
  icon: string;
  display_order: number;
}

export interface IssueResponse {
  id: string;
  message: string;
  is_admin: boolean;
  admin_name?: string;
  attachments: any[];
  created_at: string;
}

export interface Issue {
  id: string;
  subject: string;
  description: string;
  status: "open" | "in_progress" | "resolved" | "closed";
  priority: "low" | "medium" | "high" | "critical";
  category: { id: string; name: string; icon: string } | null;
  screenshot_urls: string[];
  response_count: number;
  last_response_at?: string;
  last_response_is_admin?: boolean;
  responses?: IssueResponse[];
  created_at: string;
  updated_at: string;
}

export interface IssueSummary {
  total: number;
  open: number;
  in_progress: number;
  resolved: number;
}

export const issuesApi = {
  // Issue Categories
  getCategories: () => get<IssueCategory[]>("/issues/categories"),

  // My Issues
  getMyIssues: (params?: { page?: number; limit?: number; status?: string }) =>
    get<any>(`/issues/${buildQueryString(params)}`),

  // Issue Detail
  getIssue: (issueId: string) => get<Issue>(`/issues/${issueId}`),

  // Submit Issue
  createIssue: (data: {
    category_id: string;
    subject: string;
    description: string;
    priority?: string;
    screenshot_urls?: string[];
  }) => post<{ id: string; subject: string; status: string }>("/issues/", data),

  // Reply to Issue
  replyToIssue: (issueId: string, data: { message: string }) =>
    post<{ id: string }>(`/issues/${issueId}/reply`, data),

  // Close Issue
  closeIssue: (issueId: string) => put<any>(`/issues/${issueId}/close`),
};

File: backend/app/models/issues.py
from sqlalchemy import Column, Boolean, ForeignKey, DateTime, Integer, Text, Enum
from sqlalchemy.dialects.postgresql import UUID, JSONB
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from core.base import Base
from models.enums import IssueStatusEnum, IssuePriorityEnum


class IssueCategory(Base):
    __tablename__ = 'issue_categories'

    id = Column(UUID(as_uuid=True), primary_key=True, server_default=func.gen_random_uuid())
    name = Column(Text, nullable=False, unique=True)
    description = Column(Text)
    icon = Column(Text)
    display_order = Column(Integer, default=0)
    is_active = Column(Boolean, default=True)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

    # Relationships
    issues = relationship("Issue", back_populates="category")


class Issue(Base):
    __tablename__ = 'issues'

    id = Column(UUID(as_uuid=True), primary_key=True, server_default=func.gen_random_uuid())
    user_id = Column(UUID(as_uuid=True), ForeignKey('users.id', ondelete='CASCADE'), nullable=False)
    category_id = Column(UUID(as_uuid=True), ForeignKey('issue_categories.id', ondelete='RESTRICT'), nullable=False)
    subject = Column(Text, nullable=False)
    description = Column(Text, nullable=False)
    status = Column(Enum(IssueStatusEnum, name="issue_status_enum"), default=IssueStatusEnum.open)
    priority = Column(Enum(IssuePriorityEnum, name="issue_priority_enum"), default=IssuePriorityEnum.medium)
    screenshot_urls = Column(JSONB, server_default="'[]'::jsonb")
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

    # Relationships
    user = relationship("User", back_populates="issues")
    category = relationship("IssueCategory", back_populates="issues")
    responses = relationship("IssueResponse", back_populates="issue", order_by="IssueResponse.created_at.asc()")


class IssueResponse(Base):
    __tablename__ = 'issue_responses'

    id = Column(UUID(as_uuid=True), primary_key=True, server_default=func.gen_random_uuid())
    issue_id = Column(UUID(as_uuid=True), ForeignKey('issues.id', ondelete='CASCADE'), nullable=False)
    responder_id = Column(UUID(as_uuid=True))
    is_admin = Column(Boolean, default=False)
    admin_name = Column(Text)
    message = Column(Text, nullable=False)
    attachments = Column(JSONB, server_default="'[]'::jsonb")
    created_at = Column(DateTime, server_default=func.now())

    # Relationships
    issue = relationship("Issue", back_populates="responses")
